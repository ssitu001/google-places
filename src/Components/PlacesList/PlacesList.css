.places-content {
  height: calc(100vh - 200px);
  flex: 1;
  overflow-y: scroll;
  cursor: pointer;
}