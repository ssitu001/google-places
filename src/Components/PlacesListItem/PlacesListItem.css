.list-group-item-alternate {
  background: #ececec;
}