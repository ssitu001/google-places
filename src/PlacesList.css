.places-content {
  height: 500px;
  flex: 1;
  overflow-y: scroll;
  cursor: pointer;
}