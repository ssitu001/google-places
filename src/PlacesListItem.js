import React from 'react';

const PlacesListItem = ({place, index, displayInfoFromListItem}) => {

  return (
    <div className="list-group-item" onClick={() => displayInfoFromListItem(index)}>
      <h5 className="list-group-item-heading">
        {place.name}
      </h5>
      <p>{place.formatted_address}</p>
    </div>
  )
}

export default PlacesListItem;