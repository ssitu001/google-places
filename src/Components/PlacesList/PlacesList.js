import React from 'react';

import PlacesListItem from '../PlacesListItem/PlacesListItem';

import './PlacesList.css';

const PlacesList = ({places, displayInfoFromListItem}) => {

  return (
    <div className="list-group places-content">
      {places.length ? places.map((place, i) =>
        <PlacesListItem 
          key={i} 
          index={i} 
          place={place} 
          displayInfoFromListItem={displayInfoFromListItem}
        />
      ) : null}
    </div>
  )
}

export default PlacesList;