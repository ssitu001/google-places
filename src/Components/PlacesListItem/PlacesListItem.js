import React from 'react';

import './PlacesListItem.css';

const PlacesListItem = ({place, index, displayInfoFromListItem}) => {

  return (
    <div 
      className={`list-group-item ${index % 2 === 0 ? 'list-group-item-alternate' : ''}`}
      onClick={() => displayInfoFromListItem(index)}>
      <h5 className="list-group-item-heading">
        {place.name}
      </h5>
      <p>{place.formatted_address}</p>
    </div>
  )
}

export default PlacesListItem;