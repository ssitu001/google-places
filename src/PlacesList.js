import React from 'react';

import PlacesListItem from './PlacesListItem';

const PlacesList = (props) => {
  
  const placesList = props.places.map((place, i) =>
    <PlacesListItem key={i} place={place}/>
  );

  return (
    <div>
      {placesList}
    </div>
  )
}

export default PlacesList;