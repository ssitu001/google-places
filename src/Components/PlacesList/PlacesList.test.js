import React from 'react';
import toJson from 'enzyme-to-json';

import PlacesList from './PlacesList';
import PlacesListItem from '../PlacesListItem/PlacesListItem';

test('renders PlacesList component with places', () => {
  const places = [{
    name: 'Place1',
    formatted_address: '123 Fake St',
  }, {
    name: 'Place2',
    formatted_address: '456 Fake St',
  }];
  const component = shallow(<PlacesList places={places}/>);
  const tree = toJson(component);

  expect(tree).toMatchSnapshot();
  expect(component.find(PlacesListItem).length).toBe(2);
  expect(component).toHaveLength(1);
});

test('renders PlacesList component with no places', () => {
  const places = [];
  const component = shallow(<PlacesList places={places}/>);
  const tree = toJson(component);
  
  expect(tree).toMatchSnapshot();
  expect(component.find(PlacesListItem).length).toBe(0);
  expect(component).toHaveLength(1);
});