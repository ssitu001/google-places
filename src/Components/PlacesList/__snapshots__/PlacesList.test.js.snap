// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders PlacesList component with no places 1`] = `
<div
  className="list-group places-content"
/>
`;

exports[`renders PlacesList component with places 1`] = `
<div
  className="list-group places-content"
>
  <PlacesListItem
    index={0}
    key="0"
    place={
      Object {
        "formatted_address": "123 Fake St",
        "name": "Place1",
      }
    }
  />
  <PlacesListItem
    index={1}
    key="1"
    place={
      Object {
        "formatted_address": "456 Fake St",
        "name": "Place2",
      }
    }
  />
</div>
`;
